lib/astutils.cpp

```diff
@@ -2806,7 +2806,8 @@ Token* findVariableChanged(Token *start, const Token *end, int indirect, const n
     if (depth < 0)
         return start;
     auto getExprTok = memoize([&] {
-        return findExpression(start, exprid);
+        const auto* e = findExpression(start, exprid);
+        return e ? e : start;
```

Comment:
It shouldn't return `start`. It supposed to return the token that has that exprid or `nullptr` if it can't be found.

I dont really understand how this would fix a hang though as you are still calling `findExpression`. As a perf improvement we could probably skip searching of a variable changed if its already a unique expression id.

Reply:
Ah sorry, I see your other comment on why you did this, but it still shouldn't return `start` as that is wrong.